src/pathtree.ts

```diff
@@ -198,4 +161,52 @@ export module PathTree {
 
     return [null, tree];
   }
+
+  /**
+   * @description Compress nested trees for smaller output
+   */
+  export function compress<T>(tree: Tree<T | null>) {
+    if (tree.length === 1) {
+      // tree root is Leaf
+      return;
+    }
+
+    const origin = tree[1];
+    const keys = Object.keys(origin).sort((a, b) => a.length - b.length);
```

Comment:
It's important that parent keys are processed before their children. 
In the first commit this only was for performance reasons, but now this is actually crucial. If a child-key would be added to the `transformed` before it's parent, it wouldn't be detected as a child afterwards. (again for performance reasons)
As every child has a longer key than it's parents, sorting by length is the easiest way to ensure this works as expected.

Reply:
I see. You'd really want to count the number of `.` then, right?

Reply:
Now that I think about it more, sorting by length also guarantees that. Nice!

Reply:
None of our tests fail if we just omit the sorting altogether. Maybe our trees are always in the right order, anyway? I'll try to create to disprove that.

Reply:
I think they are. (For example: If we detect a Map, we add the Map statement first and the Map-children afterwards.) But I don't know if `Object.keys` guarantees a specific order. I don't think so. Therefore it would be more safe to sort the keys. It does not take much time, so there should be no reason to omit sorting.

Reply:
"It does not take much time" - well, for big instances, it does.

But: https://stackoverflow.com/a/280861 states that iteration order isn't well-defined in the JS standard, so I guess we'll have to sort 😅